xbmc/addons/interfaces/General.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "General.h"
+
+#include "addons/kodi-addon-dev-kit/include/kodi/General.h"
+
+#include "Application.h"
+#include "addons/AddonDll.h"
+#include "addons/GUIDialogAddonSettings.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "utils/CharsetConverter.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
+
+using namespace kodi; // addon-dev-kit namespace
+
+namespace ADDON
+{
+
+void Interface_General::Init(AddonGlobalInterface* addonInterface)
+{
+  addonInterface->toKodi->kodi = (AddonToKodiFuncTable_kodi*)malloc(sizeof(AddonToKodiFuncTable_kodi));
+
+  addonInterface->toKodi->kodi->open_settings_dialog = open_settings_dialog;
+  addonInterface->toKodi->kodi->get_localized_string = get_localized_string;
+  addonInterface->toKodi->kodi->unknown_to_utf8 = unknown_to_utf8;
+  addonInterface->toKodi->kodi->queue_notification = queue_notification;
+  addonInterface->toKodi->kodi->queue_notification_from_type = queue_notification_from_type;
+  addonInterface->toKodi->kodi->queue_notification_with_image = queue_notification_with_image;
```

Comment:
On addon itself you have only "QueueNotification" but for the "C" must be another name for the types.

With the different ways of Kodi becomes it hard to hold easy for addons, hope the docu can help addon developers http://alwinesch.github.io/group__cpp__kodi.html#gacb09570070a6a601974ecb76ab9e521b